utils/zerovec/src/map/mod.rs

```diff
@@ -331,3 +332,48 @@ where
         }
     }
 }
+
+// We can't use the default PartialEq because ZeroMap is invariant
+// so otherwise rustc will not automatically allow you to compare ZeroMaps
+// with different lifetimes
+impl<'a, 'b, K, V> PartialEq<ZeroMap<'b, K, V>> for ZeroMap<'a, K, V>
+where
+    K: for<'c> ZeroMapKV<'c>,
+    V: for<'c> ZeroMapKV<'c>,
```

Comment:
Should these also be `?Sized`? Adding that bound gets my code slightly closer to compiling.

Reply:
oh, yes.